test/over_react_component_test.html

```diff
@@ -21,6 +21,7 @@
     <!-- javascript -->
     <script src="packages/react/react_with_addons.js"></script>
     <script src="packages/react/react_dom.js"></script>
+    <script src="packages/resize_observer_polyfill_wrapper/resize_observer_polyfill.js"></script>
```

Comment:
Is this JS file necessary if `props.useResizeObserver` is `true`?

Is there any sort of validation within the component enforcing this?

If not - at the very least - there should be extensive documentation in the doc comment for `props.useResizeObserver` explaining that this is a requirement, IMO.

Reply:
It's only necessary if that is `true`.  There's no validation here, just my underlying [package](https://github.com/JaceHensley/resize_observer_polyfill_wrapper)

Reply:
So, would we need to include this JS file in the main shell .html file? Hopefully with feature detection and only load it if it needs the polyfill?

Reply:
The JS library that I wrapped never uses the native ResizeObserver, for whatever reason, so this JS file will always need to be loaded.

Reply:
oh, that's weird. Isn't the whole point of using the browser built in resize observer to reduce code? This change just replaces the resize detection logic from our Dart polyfill to a JS implemented polyfill then?

Reply:
Kinda, but it's a lot cleaner of an impl IMO. It doesn't wrap the resizing element and place "resize sensors" around it. And I _think_ it's more performant. I want to try it out with comments and report back on that.

Reply:
@jacehensley-wf IMO - if this script is necessary for the component to work when `props.useResizeObserver` is true, then the component should be responsible for dynamically loading / unloading that script into the DOM.

Reply:
But how does it know where the script is?

Reply:
You could load it as a resource and inject it as a script?
See https://github.com/dart-lang/resource

Reply:
Interesting, I'll take a stab at that tonight

Reply:
hmm I guess that makes more sense to do in the wrapper dart package

Reply:
The more I'm thinking about it the less I like the approach of injecting js without the user doing it. I think throwing an error and letting the user add the js file is the best approach.